### ISSUE TITLE
"loginctl lock-session" fails when no explicit session passed

### ISSUE THREAD

**Issue by tchernomax** [systemd/systemd#13614]
**systemd version the issue has been seen with**

```
systemd 243 (243.0-1-arch)
+PAM +AUDIT -SELINUX -IMA -APPARMOR +SMACK -SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD +IDN2 -IDN +PCRE2 default-hierarchy=hybrid
```

**Used distribution**

archlinux

**Expected behaviour you didn't see**

```
max@mde-oxalide % loginctl lock-session
# session locked
```

**Unexpected behaviour you saw**

```
max@mde-oxalide % loginctl lock-session
Failed to issue method call: Caller does not belong to any known session.
```

**Note**

XDG_SESSION_ID isn't set ; it doesn't change anything to export it before calling `loginctl lock-session`.

Calling `loginctl lock-session X`, with `X` my session id, works.

**Comment by tchernomax** [comment]
note, direct call to dbus works:
```
busctl call org.freedesktop.login1 /org/freedesktop/login1/session/auto org.freedesktop.login1.Session Lock
```

**Comment by sepria841** [comment]
d10b1206c951e36cfa1a02684ac370a0b8e809d08959d6d78d002c74f6c4cf3c

**Comment by tchernomax** [comment]
@sepria841 ?

**Comment by poettering** [comment]
So from which context are you invoking this? How does your systemd-cgls look like, i.e. to which cgroup precisely does the shell you invoking this in belong to?

**Comment by tchernomax** [comment]
sorry for the late response.

```
max@mde-oxalide % loginctl lock-session
Failed to issue method call: Caller does not belong to any known session.
max@mde-oxalide % systemctl status $$
● user@1000.service - User Manager for UID 1000
   Loaded: loaded (/usr/lib/systemd/system/user@.service; static; vendor preset: disabled)
   Active: active (running) since Thu 2019-11-21 15:01:45 CET; 2h 46min ago
     Docs: man:user@.service(5)
 Main PID: 951 (systemd)
   Status: "Startup finished in 81ms."
    Tasks: 747
   Memory: 7.9G
   CGroup: /user.slice/user-1000.slice/user@1000.service
           …
```

I am on a gnome/wayland session on a uptodate archlinux.

**Comment by poettering** [comment]
Fix waiting in #14626
